docs/contributing/coding-guidelines/ros-nodes/directory-structure.md

```diff
@@ -1,75 +1,211 @@
+- The package name should be entered within the `<name>` tag.
+  - `<name>autoware_gnss_poser</name>`
 
-Rationale: Since ROS parameters files are type-sensitive, they should not be the target of some code formatters and linters. In order to distinguish the file type, we use different file extensions.
+#### `CMakeLists.txt`
 
-#### `doc`
+- The [`project()`](https://cmake.org/cmake/help/latest/command/project.html) command should call the package name.
+  - **Example:** `project(autoware_gnss_poser)`
 
-Place document files and link from README.
+##### Exporting a component
 
-#### `include`
+```cmake
+ament_auto_add_library(${PROJECT_NAME} SHARED
+  src/gnss_poser_node.cpp
+)
 
-Place header files exposed to other packages. Do not place files directly under the `include` directory, but place files under the directory with the package name.
-This directory is used for mostly library headers. Note that many headers do not need to be placed here. It is enough to place the headers under the `src` directory.
+rclcpp_components_register_node(${PROJECT_NAME}
+  PLUGIN "autoware::gnss_poser::GNSSPoserNode"
+  EXECUTABLE ${PROJECT_NAME}_node
+)
+```
 
-Reference: <https://docs.ros.org/en/humble/How-To-Guides/Ament-CMake-Documentation.html#adding-targets>
+- The component executable should have `_node` suffix.
+- The component executable name should start with `${PROJECT_NAME}`
 
-#### `launch`
+### `config` and `schema`
 
-Place launch files (`.launch.xml` and `.launch.py`).
+```txt
+autoware_gnss_poser
+│─ config
+│   ├─ gnss_poser.param.yaml
+│   └─ another_non_ros_config.yaml
+└─ schema
+    └─ gnss_poser.schema.json
+```
+
+#### `config`
+
+- ROS parameters uses the extension `.param.yaml`.
+- Non-ROS parameters use the extension `.yaml`.
+
+**Rationale:** Different linting rules are used for ROS parameters and non-ROS parameters.
 
 #### `schema`
 
-Place parameter definition files. See [parameters](./parameters.md) for details.
+Place parameter definition files. See [Parameters](./parameters.md) for details.
 
-#### `src`
+### `doc`
 
-Place source files and private header files.
+```txt
+autoware_gnss_poser
+└─ doc
+    ├─ foo_document.md
+    └─ foo_diagram.svg
+```
 
-#### `test`
+Place documentation files and link them from the README file.
+
+### `include` and `src`
+
+- Unless you specifically need to export headers, you shouldn't have a `include` directory under the package directory.
+- For most cases, follow [Not exporting headers](#not-exporting-headers).
+- Library packages that export headers may follow [Exporting headers](#exporting-headers).
+
+#### Not exporting headers
+
+```txt
+autoware_gnss_poser
+└─ src
+    ├─ include
+    │   ├─ gnss_poser_node.hpp
+    │   └─ foo.hpp
+    │─ gnss_poser_node.cpp
+    └─ bar.cpp
+```
+
+- Put the header files in the `include` directory under the `src` directory.
+- The source file exporting the node should:
+  - have `_node` suffix.
+    - **Rationale:** To distinguish from other source files.
```

Comment:
I think the maintainer should be free to decide on the structure below src. I don't think src/include is necessary. For simple nodes, it is simpler to arrange them in the same hierarchy. There is also an alternative method for `_node` suffix, such as dividing it by directory structure.

Reply:
I understand, we can leave the `include` folder under `src` as optional.

Sure, they can arrange the folder structure as they wish.

But I think it is easier to enforce `_node` suffix for the source files that are exporting components or for nodes, don't you think?

Reply:
Allowed devs to rearrange inside src with: https://github.com/autowarefoundation/autoware-documentation/pull/547/commits/cede324ceeb2a45dbe36f68cb75feec0566b847b

But kept the `_node` suffix, waiting for your comments.

Reply:
> But I think it is easier to enforce _node suffix for the source files that are exporting components or for nodes, don't you think?

That make sense about node suffix from this comment. https://github.com/autowarefoundation/autoware-documentation/pull/547#discussion_r1595177989